src/config.ts

```diff
@@ -313,3 +307,70 @@ export const SOFTWARE_VERSION = (() => {
     // into <pre> or <code> where it will create an unnecessary newline.
     return /^(.*)$/m.exec(versionFile)?.at(0) ?? defaultText;
 })();
+
+
+// Command line related functions
+
+/**
+ * Grabs an option from the command line and checks if it exists.
+ * @param arg Option name
+ * @returns True if the option is present, otherwise false.
+ */
+function isCommandLineOptionPresent(args: string[], arg: string): boolean {
+    return args.includes(arg);
+}
+
+/**
+ * Grabs an option's value from program options if it exists, otherwise returns undefined.
+ * @param arg Option name
+ * @param throwOnInvalid If true or undefined, throws an error if the option is present but has no value. If false, returns undefined.
+ * @returns The value passed to the option, or undefined if the option is not specified.
+ * @throws Error if the option is present but has no value.
+ */
+function getCommandLineStringOption(args: string[], arg: string, throwOnInvalid?: boolean): string | undefined {
+    // We don't want to throw if the option is not present
+    if(!isCommandLineOptionPresent(args, arg)) {
+        return undefined;
+    }
+
+    const optionIndex = args.indexOf(arg);
+    if (optionIndex === -1) {
+        return undefined;
+    }
+
+    //check if the next index is not an option
+    if (args[optionIndex + 1] && !args[optionIndex + 1].startsWith("--")){
+        return args[optionIndex + 1];
+    }
+
+    if (throwOnInvalid === undefined || throwOnInvalid) {
+        throw new Error(`Invalid value provided for ${arg}`);
+    }
+
+    return undefined;
+}
+
+/**
+ * Grabs a path option from the command line and checks if it exists.
+ * @param arg Option name
+ * @param throwOnInvalid If true or undefined, throws an error if the path does not exist. If false, returns undefined.
```

Comment:
`throwOnInvalid` makes less sense here. If someone wanted to set this explicitly to false, so they could use the path as a location to write to (given we have included a check to see if the path exists here), then this function will not throw if a path wasn't provided at all.

Reply:
If it was included so that someone could provide a path to write to instead of read from, then i suggest we just remove the check for `fs.existsSync` entirely. And leave it to the caller to verify if the path exists.

Reply:
Should I invert the logic and rename it to `checkPathExists`?

Reply:
I think the logic should be removed entirely, surely we will choke if the path doesn't exist when someone tries to read it, which in our case isn't very far down the line.